xbmc/platform/android/storage/AndroidStorageProvider.cpp

```diff
@@ -414,3 +416,73 @@ bool CAndroidStorageProvider::PumpDriveChangeEvents(IStorageEventsCallback *call
   m_removableDrives = std::move(drives);
   return changed;
 }
+
+namespace
+{
+constexpr float GIGABYTES = 1073741824;
+constexpr int PATH_MAXLEN = 38;
+} // namespace
+
+bool CAndroidStorageProvider::GetStorageUsage(const std::string& path, std::string& usage)
+{
+  if (path.empty())
+  {
+    std::ostringstream fmt;
+
+    fmt.width(PATH_MAXLEN);
+    fmt << std::left << "Filesystem";
+
+    fmt.width(12);
+    fmt << std::right << "Size";
+
+    fmt.width(12);
+    fmt << "Used";
+
+    fmt.width(12);
+    fmt << "Avail";
+
+    fmt.width(12);
+    fmt << "Use %";
+
+    usage = fmt.str();
+    return false;
+  }
+
+  CJNIStatFs fileStat(path);
+  const int blockSize = fileStat.getBlockSize();
+  const int blockCount = fileStat.getBlockCount();
+  const int freeBlocks = fileStat.getFreeBlocks();
+
+  if (blockSize <= 0 || blockCount <= 0 || freeBlocks < 0)
+    return false;
+
+  const float totalSize = static_cast<float>(blockSize) * blockCount / GIGABYTES;
+  const float freeSize = static_cast<float>(blockSize) * freeBlocks / GIGABYTES;
+  const float usedSize = totalSize - freeSize;
+  const float usedPercentage = usedSize / totalSize * 100;
+
+  std::ostringstream fmt;
```

Comment:
It would be nice if instead of stringstream we could just use StringUtils::Format, the use of fmt::format is more readable.

Not required for this PR but just a suggestion.

Reply:
Updated using StringUtils::Format